src/mudlet.cpp

```diff
@@ -691,16 +694,92 @@ mudlet::mudlet()
 //    });
 }
 
-QSettings* mudlet::getQSettings()
+static QString findExecutableDir() {
+    // Linux AppImage support
+    QProcessEnvironment systemEnvironment = QProcessEnvironment::systemEnvironment();
+    if (systemEnvironment.contains(qsl("APPIMAGE"))) {
+        QString appimgPath = systemEnvironment.value(qsl("APPIMAGE"), QString());
+        return QFileInfo(appimgPath).dir().path();
+    }
+    return QCoreApplication::applicationDirPath();
+}
+
+static QString readMarkerFile(const QString& path) {
+    QString line;
+    QFile file(path);
+    file.open(QIODevice::ReadOnly);
+    QTextStream(&file).readLineInto(&line);
+    file.close();
+    return line;
+}
+
+static bool validateConfDir(QString& path) {
+    if (path.size() == 0) {
+        qWarning("WARN: portable data path not specified");
+        return false;
+    }
+    QFileInfo pathInfo(path);
+    if (pathInfo.isFile()) {
+        qWarning("WARN: specified portable data path is an existing file: %s", qPrintable(path));
```

Comment:
Using `QWarning`/`QDebug`/`QInfo` for these messages may not be the best option here as they may not show up for all users - especially if they are not running Mudlet from a terminal window. For these particular cases it might be appropriate to use the normally discouraged by us (Core Devs) `QMessageBox` - as it is, this sort of problem is sufficient to stop the end-user using Mudlet in the manner they wish so having to stop and read a "modal" message could, I think, be tolerated. @vadi2 your thoughts on this?

Reply:
Also, these messages are not going to be translated so could be problematic for non-English users...

Reply:
Having had supported all sorts of users for two decades now, I reckon this warning is OK and better than a pop-up.

In case someone is active trying to make Mudlet portable work and it's not working, we can advise them to launch from the terminal to see any issues.

In case someone non-technical isn't trying to make Mudlet portable work though, and they get blocking pop ups, it's always a bit of a disaster as people get confused / frustrated / upset and don't know what to do.

So this approach has the least problems.